File: api/src/main/java/org/axostudio/axohologram/api/hologram/VisibilityMode.java
package org.axostudio.axohologram.api.hologram;

import java.util.Locale;

public enum VisibilityMode {
    ALL,
    PERMISSION,
    CONDITION,
    MANUAL,
    SCRIPT,
    RADIUS,
    NONE;

    public static VisibilityMode fromString(String raw) {
        if (raw == null || raw.trim().isEmpty()) {
            return ALL;
        }
        try {
            return valueOf(raw.trim().toUpperCase(Locale.ROOT));
        } catch (IllegalArgumentException e) {
            return ALL;
        }
    }
}

File: src/main/kotlin/org/axostudio/axohologram/core/hologram/visibility/VisibilityMode.kt
package org.axostudio.axohologram.core.hologram.visibility

import java.util.Locale

enum class VisibilityMode {
    ALL,
    PERMISSION,
    CONDITION,
    MANUAL,
    SCRIPT,
    RADIUS,
    NONE;

    companion object {
        @JvmStatic
        fun fromString(raw: String?): VisibilityMode {
            if (raw.isNullOrBlank()) return ALL
            return runCatching {
                valueOf(raw.trim().uppercase(Locale.ROOT))
            }.getOrDefault(ALL)
        }
    }
}
